File: leetcode-in-python3/solution/136_single_number.py
from typing import List

class Solution:
    def romanToInt(self, s: str) -> int:
        t = {
            "I": 1,
            "V": 5,
            "X": 10,
            "L": 50,
            "C": 100,
            "D": 500,
            "M": 1000,
        }
        ans = 0
        i = 0
        while i < len(s):
            if i + 1 < len(s) and t[s[i]] < t[s[i+1]]:
                ans += t[s[i+1]] - t[s[i]]
                i += 2
            else:
                ans += t[s[i]]
                i += 1
        return ans

File: leetcode-in-rust/src/solution/_136_single_number.rs
pub struct Solution {}

impl Solution {
    pub fn single_number(nums: Vec<i32>) -> i32 {
        let mut r = 0;
        for &i in nums.iter() {
            r ^= i;
        }
        r
    }
}


#[cfg(test)]
mod tests {
    use super::*;


    #[test]
    fn test_solution() {
        let test_case = vec![
            (vec![2, 2, 1], 1),
            (vec![4, 1, 2, 1, 2], 4),
        ];
        for (nums, answer) in test_case {
            let ans = Solution::single_number(nums);
            assert_eq!(answer, ans);
        }
    }
}